zebra/zebra_rib.c

```diff
@@ -1748,6 +1780,256 @@ static void rib_process(struct route_node *rn)
 	rib_gc_dest(rn);
 }
 
+/*
+ * Utility to match route with dplane context data
+ */
+static bool rib_route_match_ctx(const struct route_entry *re,
```

Comment:
Why not assign a uint64_t context value to each route_entry and use that to pass back and forth instead of this complicated dance?